src/goto-instrument/contracts/assigns.cpp

```diff
@@ -110,32 +110,57 @@ void assigns_clauset::add_target(const exprt &target_expr)
   }
 }
 
-void assigns_clauset::remove_target(const exprt &target_expr)
+void assigns_clauset::remove_global_write_set(const exprt &target_expr)
 {
-  targets.erase(targett(*this, targett::normalize(target_expr)));
+  global_write_set.erase(targett(*this, target_expr));
+}
+
+void assigns_clauset::add_local_write_set(const exprt &expr)
+{
+  auto insertion_succeeded = local_write_set.emplace(*this, expr).second;
+
+  if(!insertion_succeeded)
+  {
+    log.warning() << "Ignored duplicate expression '"
+                  << from_expr(ns, expr.id(), expr) << "' in assigns clause at "
```

Comment:
This warning message can be a little confusing, since the user-specified targets only appear within `global_write_set`.

Should we just remove this warning and silently drop any local targets that we might internally add accidentally? Or, we could fatally fail on an `INVARIANT` failure here if we want to be absolutely sure that we never double-add local targets, but I think that might be too aggressive.

Reply:
Implemented as suggested.